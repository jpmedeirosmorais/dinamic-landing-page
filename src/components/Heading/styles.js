import styled from 'styled-components';
import { css } from 'styled-components';

export const Title = styled.h1`
  ${({ theme, light }) => css`
    color: ${light ? theme.colors.primary : theme.colors.white};
    font-family: ${theme.fonts.family.montserrat};
  `}
`;
