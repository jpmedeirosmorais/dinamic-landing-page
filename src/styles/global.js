import { css } from 'styled-components';
import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }
  html {
    font-size: 62.5%; // 16px = 1.6rem
  }
  body {
    -webkit-font-smoothing: antialiased;
  }
  body, input, textarea, button {
    font-family: 'Inter',sans-serif;
    font-weight: 1.6rem;
  }
  h1, h2, h3, h4, h5, h6, strong {
    font-weight: 600;
  }
  span {
    font-size: 1.6rem;
  }
  button {
    cursor: pointer;
  }
  [disabled] {
    opacity: 0.6;
    cursor: not-allowed;
  }
`;
