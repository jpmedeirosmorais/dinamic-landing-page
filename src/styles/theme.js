export const theme = {
  colors: {
    primary: '#0A1128',
    secondary: '#DC143C',
    white: '#FFFFFF',
    black: '#000000',
    gray: '#EAEAEA',
    lightGray: '#F5F5F5',
    darkGray: '#BDBDBD',
  },
  fonts: {
    family: {
      sans: "'Open Sans', sans-serif",
      montserrat: "'Montserrat', sans-serif",
    },
  },
  spacings: {},
};
