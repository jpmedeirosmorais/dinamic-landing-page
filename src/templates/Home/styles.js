import styled from 'styled-components';
import { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.mainBg};
  `}
`;
