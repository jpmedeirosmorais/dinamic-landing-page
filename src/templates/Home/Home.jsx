import * as S from './styles';

function Home() {
  return (
    <S.Wrapper>
      <h1>Hello</h1>
    </S.Wrapper>
  );
}

export default Home;
