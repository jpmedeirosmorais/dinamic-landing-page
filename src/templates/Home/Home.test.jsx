import { screen } from '@testing-library/react';
import Home from './Home';
import { renderTheme } from '../../styles/render-theme';

test('renders learn react link', () => {
  const { debug } = renderTheme(<Home />);
  const headingContainer = screen.getByRole('heading', {
    name: 'Hello',
  }).parentElement;
  expect(headingContainer).toHaveStyle({
    backgroundColor: 'red',
  });
  expect(headingContainer).toMatchSnapshot();
  expect(headingContainer).toHaveStyleRule('background-color', 'red');
});
