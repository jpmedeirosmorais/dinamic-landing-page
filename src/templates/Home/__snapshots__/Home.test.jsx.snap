// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
.c0 {
  background-color: red;
}

<div
  class="c0"
>
  <h1>
    Hello
  </h1>
</div>
`;
